### PR TITLE
Rianfowler/sc 49163/do not show redeploy nor deploy buttons in

### PR DESCRIPTION
**Pull request by rianfowler**
#### Which issue(s) this PR fixes:
[sc-49163]
[sc-50832]

#### Special notes for your reviewer:
- needs to have helm managed mode on. see readme for instructions 

#### Does this PR introduce a user-facing change?
None- everything is behind a feature flag 

#### Does this PR require documentation?
NONE

## Loom
https://www.loom.com/share/da881a7912b341248cd07152fd007e50

### CODE REVIEW

File: web/src/components/apps/AppConfig.jsx [hunk 1]
```diff
@@ -513,22 +516,77 @@ class AppConfig extends Component {
...
-              <div className="u-paddingBottom--30">
-                <Loader size="30" />
-              </div>
-              :
-              <div className="ConfigError--wrapper flex-column u-paddingBottom--30 alignItems--flexStart">
-                {configError && <span className="u-textColor--error u-marginBottom--20 u-fontWeight--bold">{configError}</span>}
-                <button className="btn primary blue" disabled={!changed && !fromLicenseFlow || this.isConfigReadOnly(app)} onClick={this.handleSave}>{fromLicenseFlow ? "Continue" : "Save config"}</button>
-              </div>
-            }
+            <IsHelmManaged>
```
Comment (rianfowler) [inline web/src/components/apps/AppConfig.jsx:519]: This is a WIP- will refactor `AppConfig` into a functional component so we don't have to use render props.

File: web/src/components/apps/AppConfig.jsx [hunk 2]
```diff
@@ -513,22 +516,77 @@ class AppConfig extends Component {
...
+              {({ isHelmManaged }) => {
+                const { saveConfig, isSaving, error } = useSaveConfig({
```
Comment (rianfowler) [inline web/src/components/apps/AppConfig.jsx:521]: This hook saves the config. I didn't remove the old way of saving the config yet- this is only used by the new helm modal.

File: web/src/components/apps/AppConfig.jsx [hunk 3]
```diff
@@ -513,22 +516,77 @@ class AppConfig extends Component {
...
+                  appSlug: this.getSlug(),
+                });
+                const {
+                  download,
+                  error: downloadError,
+                  isDownloading,
+                  name,
+                  ref,
+                  url,
+                } = useDownloadValues({
```
Comment (rianfowler) [inline web/src/components/apps/AppConfig.jsx:531]: the values.yaml is added to the dom as a blob and downloaded.

File: web/src/components/apps/AppConfig.jsx [hunk 4]
```diff
@@ -513,22 +516,77 @@ class AppConfig extends Component {
...
+                    </div>}
+                    {!isHelmManaged && savingConfig &&
+                      <div className="u-paddingBottom--30">
+                        <Loader size="30" />
+                      </div>}
+                    {!isHelmManaged && !savingConfig &&
+                      <div className="ConfigError--wrapper flex-column u-paddingBottom--30 alignItems--flexStart">
+                        {configError && <span className="u-textColor--error u-marginBottom--20 u-fontWeight--bold">{configError}</span>}
+                        <button className="btn primary blue" disabled={!changed && !fromLicenseFlow || this.isConfigReadOnly(app)} onClick={this.handleSave}>{fromLicenseFlow ? "Continue" : "Save config"}</button>
+                      </div>}
```
Comment (rianfowler) [inline web/src/components/apps/AppConfig.jsx:568]: Do not show the save buttons when in helm managed mode

File: web/src/components/apps/DashboardVersionCard.jsx [hunk 5]
```diff
@@ -454,6 +455,12 @@ class DashboardVersionCard extends React.Component {
   }
 
   deployVersion = (version, force = false, continueWithFailedPreflights = false, redeploy = false) => {
+    if (this.props.isHelmManaged) {
```
Comment (rianfowler) [inline web/src/components/apps/DashboardVersionCard.jsx:458]: if in helm managed mode then we show the helm managed modal and do nothing else.

File: web/src/components/hooks/useDownloadValues.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      setError(error);
+    }
+  };
+
+  return {
+    download,
+    error,
+    isDownloading,
+    name,
+    ref,
```
Comment (rianfowler) [inline web/src/components/hooks/useDownloadValues.js:67]: the parent component provides a ref to an anchor tag

File: web/src/components/hooks/useIsHelmManaged.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import _fetch from "isomorphic-fetch";
```
Comment (alicenstar) [inline web/src/components/hooks/useIsHelmManaged.js:1]: Is this used anywhere in this file?

File: web/src/scss/utilities/modals.scss [hunk 8]
```diff
@@ -77,34 +90,99 @@
...
+  color: #000;
+}
+
+.subtitle code {
+  // padding: 2px 3px;
+  background-color: #4A4A4A;
+  overflow: auto;
+  // border-radius: 0.3em;
+  color: #f8f8f2;
+  // font-family: Consolas, Monaco, "Andale Mono", "Ubuntu Mono", monospace;
```
Comment (alicenstar) [inline web/src/scss/utilities/modals.scss:173]: Can we remove the commented styles here?

**Review by alicenstar — COMMENTED** [review]
It looks like a large green check icon is missing from the spritesheet, this is probably something I manually added that wasn't in the Figma version. I can try to fix that tomorrow if you'd like.

File: web/src/components/hooks/useIsHelmManaged.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+} = {}) {
+  const [isHelmManaged, setIsHelmManaged] = useState(null);
+  const [isHelmManagedLoading, setIsHelmManagedLoading] = useState(false);
+
+  useEffect(() => {
+    if (isHelmManaged === null) {
+      setIsHelmManaged(false);
+      setIsHelmManagedLoading(true);
+      _fetchIsHelmManaged()
+        .then(({ isHelmManaged }) => {
```
[the remark this answers is not in the document]
Comment (rianfowler) [reply web/src/components/hooks/useIsHelmManaged.js:41]: Think i'm going to work on fixing the linting in kots today. if it doesn't catch no-shadowed-variables then it's probably not catching a lot.

**Review by alicenstar — APPROVED** [review]

**Pull request merged**